src/shared/enrollment_op.c

```diff
@@ -477,6 +485,25 @@ static void w_enrollment_verify_ca_certificate(const SSL *ssl, const char *ca_ce
     }
 }
 
+/**
+ * @brief Concats the current key of the agent, if exist, as  part of the enrollment message
+ *
+ * @param buff buffer where the KEY section will be concatenated
+ * @param key key will be added
+ */
+static void w_enrollment_concat_key(char *buff, keyentry* key_entry) {
+    assert(buff != NULL);
+    assert(key_entry != NULL);
+
+    os_sha1 output;
+    char* opt_buf = NULL;
+    os_calloc(OS_SIZE_65536, sizeof(char), opt_buf);
```

Comment:
I suggest using a smaller size of memory. Probably `OS_SIZE_512` or `OS_SIZE_1024` would be enough.

Reply:
As we know the exact length of os_sha1 with sizeof(os_sha1) we can calculate exactly how much memory it will cost.
For code clarity, we can use OS_SIZE_512 as Diego suggested,

Reply:
Fixed

Reply:
I also think that the variable size is much bigger than necessary, but there are also many places around this code where the memory allocated is huge and we could review them later.